services/core-api/app/api/mines/work_information/models/mine_work_information.py

```diff
@@ -0,0 +1,99 @@
+from datetime import datetime
+from pytz import timezone
+
+from sqlalchemy.dialects.postgresql import UUID
+from sqlalchemy.schema import FetchedValue
+from sqlalchemy.ext.hybrid import hybrid_property
+from sqlalchemy.orm import validates
+
+from app.api.utils.models_mixins import SoftDeleteMixin, AuditMixin, Base
+from app.api.utils.include.user_info import User
+from app.extensions import db
+
+
+class MineWorkInformation(SoftDeleteMixin, AuditMixin, Base):
+    __tablename__ = 'mine_work_information'
+
+    mine_work_information_id = db.Column(
+        db.Integer, primary_key=True, server_default=FetchedValue())
+    mine_work_information_guid = db.Column(
+        UUID(as_uuid=True), server_default=FetchedValue(), nullable=False)
+    mine_guid = db.Column(UUID(as_uuid=True), db.ForeignKey('mine.mine_guid'), nullable=False)
+
+    work_start_date = db.Column(db.Date)
+    work_stop_date = db.Column(db.Date)
+    work_comments = db.Column(db.String)
+    mine_work_status_code = db.Column(db.String,
+                                      db.ForeignKey('mine_work_status.mine_work_status_code'))
+
+    deleted_ind = db.Column(db.Boolean, server_default=FetchedValue(), nullable=False)
+
+    created_by = db.Column(db.String, default=User().get_user_username, nullable=False)
+    created_timestamp = db.Column(db.DateTime, default=datetime.utcnow, nullable=False)
+    updated_by = db.Column(db.String, default=User().get_user_username, nullable=False)
+    updated_timestamp = db.Column(db.DateTime, default=datetime.utcnow, nullable=False)
+
+    def __repr__(self):
+        return f'<{self.__class__.__name__} {self.mine_work_information_id}>'
+
+    def save(self):
+        self.updated_by = User().get_user_username()
+        self.updated_timestamp = datetime.utcnow()
+        self.mine_work_status_code = self.determine_mine_work_status()
+        super(MineWorkInformation, self).save()
+
+    def determine_mine_work_status(self):
+        today = datetime.now(tz=timezone('US/Pacific')).date()
```

Comment:
UTC?

Reply:
When users set the start/stop dates they are setting the date in pacific time so we should grab "today" as the current date in pacific time and not UTC.

Reply:
The BE and should not be tied to a user's location, it is FE responsibility. FE can send timezone on convert time to UTC so BE will be able to handle it properly no matter what region request is coming from

Reply:
I actually have to revert a lot of my code back to my original implementation. I can't remember why exactly I changed it but the work status has to be a hybrid property and not a normal property. Will talk about it in the morning